From f43fa770a7a836e8b7230d0ff03b5c126213c434 Mon Sep 17 00:00:00 2001
From: Pixel <pixel@n4.mandriva.com>
Date: Wed, 28 Jan 2009 14:12:30 +0100
Subject: [PATCH] fix checking libltdl is installed/installable

the macro is used when checking wether libtldl bundled in a program should be
installed on the system. It first checks /usr/lib/libltdl.la which is wrong on
x86_64, but also wrong in case libtldl.la is not there but libltdl.so is
(since libltdl.la is not compulsory, for example fedora do not package it).

Reverting to previous behaviour inspired by AC_LIBLTDL_INSTALLABLE in
libtool.m4 of libtool-1.5.26

Signed-off-by: Pixel <pixel@n4.mandriva.com>
---
 libltdl/m4/ltdl.m4 |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/libltdl/m4/ltdl.m4 b/libltdl/m4/ltdl.m4
index f6be54a..4f5c7c4 100644
--- a/libltdl/m4/ltdl.m4
+++ b/libltdl/m4/ltdl.m4
@@ -146,21 +146,15 @@ dnl AC_DEFUN([AC_LIBLTDL_INSTALLABLE], [])
 # -----------------
 # Code shared by LTDL_INSTALLABLE and LTDL_INIT([installable]).
 m4_defun([_LTDL_INSTALLABLE],
-[if test -f $prefix/lib/libltdl.la; then
-  lt_save_LDFLAGS="$LDFLAGS"
-  LDFLAGS="-L$prefix/lib $LDFLAGS"
-  AC_CHECK_LIB([ltdl], [lt_dlinit], [lt_lib_ltdl=yes])
-  LDFLAGS="$lt_save_LDFLAGS"
+[AC_CHECK_LIB([ltdl], [lt_dlinit], [lt_lib_ltdl=yes])
   if test x"${lt_lib_ltdl-no}" = xyes; then
     if test x"$enable_ltdl_install" != xyes; then
-      # Don't overwrite $prefix/lib/libltdl.la without --enable-ltdl-install
-      AC_MSG_WARN([not overwriting libltdl at $prefix, force with `--enable-ltdl-install'])
+      AC_MSG_WARN([not overwriting installed libltdl, force with `--enable-ltdl-install'])
       enable_ltdl_install=no
     fi
   elif test x"$enable_ltdl_install" = xno; then
     AC_MSG_WARN([libltdl not installed, but installation disabled])
   fi
-fi
 
 # If configure.ac declared an installable ltdl, and the user didn't override
 # with --disable-ltdl-install, we will install the shipped libltdl.
-- 
1.6.1

